### ISSUE TITLE
Coding Standard is fairly crusty

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2217]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)| @jml reported|
|-|-|
|Trac ID|trac#2217|
|Type|defect|
|Created|2006-11-04 07:10:25Z|
|Branch|https://github.com/twisted/twisted/tree/crusty-standard-2217|

The coding standard is quite old and could use an overall edit. Some things
 * Don't recommend testing, refer to uqds
 * Don't use "I"
 * I certainly don't write code with a thesaurus open.
 * Twisted doesn't have any databases
 * `"UNDOCUMENTED"` as a docstring is never used

I'm going to just do this and then put it up for review.

<details><summary>Searchable metadata</summary>

```
trac-id__2217 2217
type__defect defect
reporter__jml jml
priority__normal normal
milestone__ 
branch__branches_crusty_standard_2217 branches/crusty-standard-2217
branch_author__thijs thijs
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation__policy documentation, policy
time__1162624225000000 1162624225000000
changetime__1237558236000000 1237558236000000
version__None None
owner__ 
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="moonfallen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_1"></a>|moonfallen commented|
|-|-|

I've always regarded the bit about dictionary and thesaurus as being metaphorical, not literal, and I think the message is still germane.  Unless you wish to refer the user to make bookmarks for dictionary.com and thesaurus.com http://dictionary.reference.com/search?q=%s and http://thesaurus.reference.com/search?config=roget&q=%s I don't think that section needs updating.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra commented|
|-|-|

It also refers to http://divmod.org/trac/wiki/BranchBasedDevelopment which is now referred to as http://divmod.org/trac/wiki/UltimateQualityDevelopmentSystem

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra commented|
|-|-|

(In [26443]) Branching to 'crusty-standard-2217'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

Replying to [jml](#2217):
>  * Don't recommend testing, refer to uqds
>  * Don't use "I"
>  * I certainly don't write code with a thesaurus open.
>  * Twisted doesn't have any databases
>  * `"UNDOCUMENTED"` as a docstring is never used
>

Addressed these points in r26445, putting it up for review.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_6"></a>|@glyph set owner to @thijstriemstra|
|-|-|

  1. TwistedDevelopment is probably more appropriate than the UQDS page.
  1. Why remove the thing about considering how you'd use the code?
  1. As long as we're updating the bit about docstrings, "whenever possible" should be replaced with something more forceful like "always" or just removed entirely.

Thanks for updating the coding standard :).

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra set owner to @glyph|
|-|-|

Replying to [glyph](#note_6):
>   1. TwistedDevelopment is probably more appropriate than the UQDS page.
>   1. Why remove the thing about considering how you'd use the code?
>   1. As long as we're updating the bit about docstrings, "whenever possible" should be replaced with something more forceful like "always" or just removed entirely.
>
> Thanks for updating the coding standard :).

Fixed these points in r26474. I removed the thing about using the code because it had a reference to 'I' and I wasn't sure how to rewrite it properly. I added it back in and rewrote it, putting it up for review again.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_8"></a>|@glyph set owner to @thijstriemstra|
|-|-|

All right.  That wording looks fine to me.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_9"></a>|@thijstriemstra set status to `assigned`|
|-|-|

Replying to [glyph](#note_8):
> All right.  That wording looks fine to me.

Cool, the doc builder is also happy: http://buildbot.twistedmatrix.com/builders/documentation/builds/348

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_10"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [26476]) Merge crusty-standard-2217: Overall edit for the coding standard

Author: thijs
Reviewer: glyph
Fixes: [#2217](https://github.com/twisted/twisted/issues/2217)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
